src/metabase/db/custom_migrations.clj

```diff
@@ -714,3 +717,176 @@
+                                                                      (seq v)
+                                                                      ;; remove nils
+                                                                      (some? v))]
+                                                          [k v]))
+                                               form))))]
+    (when (not= final-settings dashcard)
+      {:id                     id
+       :visualization_settings final-settings})))
+
+(defn- parse-to-json [& ks]
+  (fn [x]
+    (reduce #(update %1 %2 json/parse-string)
+            x
+            ks)))
+
+;; This was previously a data migration, hence the metadata. The metadata is unused but potentially useful
+;; as documentation.
+(defn- migrate-click-through!
+  {:author "dpsutton"
+   :added  "0.38.1"
+   :doc    "Migration of old 'custom drill-through' to new 'click behavior'; see #15014"}
+  []
+  (transduce (comp (map (parse-to-json :card_visualization :dashcard_visualization))
+                   (map fix-click-through)
+                   (filter :visualization_settings))
+             (completing
+              (fn [_ {:keys [id visualization_settings]}]
+                (t2/update! :model/DashboardCard id {:visualization_settings visualization_settings})))
+             nil
+             ;; flamber wrote a manual postgres migration that this faithfully recreates: see
+             ;; https://github.com/metabase/metabase/issues/15014
+             (t2/query {:select [:dashcard.id
+                                 [:card.visualization_settings :card_visualization]
+                                 [:dashcard.visualization_settings :dashcard_visualization]]
+                        :from   [[:report_dashboardcard :dashcard]]
+                        :join   [[:report_card :card] [:= :dashcard.card_id :card.id]]
+                        :where  [:or
+                                 [:like
+                                  :card.visualization_settings "%\"link_template\":%"]
+                                 [:like
+                                  :card.visualization_settings "%\"click_link_template\":%"]
+                                 [:like
+                                  :dashcard.visualization_settings "%\"link_template\":%"]
+                                 [:like
+                                  :dashcard.visualization_settings "%\"click_link_template\":%"]]})))
+
+(define-migration MigrateClickThrough
+  (migrate-click-through!))
+
+;;; Removing admin from group mapping migration
+
+(defn- raw-setting
+  "Get raw setting directly from DB.
+  For some reasons during data-migration [[metabase.models.setting/get]] return the default value defined in
+  [[metabase.models.setting/defsetting]] instead of value from Setting table."
+  [k]
+  (t2/select-one-fn :value setting/Setting :key (name k)))
```

Comment:
```suggestion
  (t2/select-one-fn :value (t2/table-name :model/Setting) :key (name k)))
```

We should change it while we're here. Like SQL migrations, these custom migrations should not be dependent on the source code, ideally, this namespace should only require low-level stuff like `jdbc, toucan2, etc`. The idea is stuff from `metabase.models.setting` can change as we go, and hooks for `:model/Setting` could be as well, so the behavior of triggering those hooks now, can be different from if we run it 2 versions into the future. so we should avoid that.